### PR TITLE
fix: matchSourceUrlPrefixes compares case insensitive

### LINKED ISSUE #10962: `matchSourceUrlPrefixes` should compare case insensitive
**How are you running Renovate?**

- [x] WhiteSource Renovate hosted app on github.com
- [ ] Self hosted

If using the hosted app, please skip to the next section.
Otherwise, if self-hosted, please complete the following:

Please select which platform you are using:

- [ ] Azure DevOps (dev.azure.com)
- [ ] Azure DevOps Server
- [ ] Bitbucket Cloud (bitbucket.org)
- [ ] Bitbucket Server
- [ ] Gitea
- [ ] github.com
- [ ] GitHub Enterprise Server
- [ ] gitlab.com
- [ ] GitLab self-hosted

Renovate version: ...

**Describe the bug**

Urls are nomally case insensive, so we should compare `matchSourceUrlPrefixes` case insensitive too.

**Relevant debug logs**

<details><summary>Click me to see logs</summary>

```
Copy/paste any log here, between the starting and ending backticks
```

</details>

**Have you created a minimal reproduction repository?**

Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.

- [ ] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [ ] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

This can happen, if repos or orgs on github are moved to lowercase only letters, see #10960

### PR DESCRIPTION
**Pull request by dario-alv**
## Changes:

matchSourceUrlPrefixes now compares as case insensitive

## Context:

Fixes #10962

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/package-rules.ts [hunk 1]
```diff
@@ -194,7 +194,7 @@ function matchesRule(
   }
   if (matchSourceUrlPrefixes.length) {
     const isMatch = matchSourceUrlPrefixes.some((prefix) =>
-      sourceUrl?.startsWith(prefix)
+      sourceUrl?.toLowerCase().startsWith(prefix.toLowerCase())
```
Comment (viceice) [inline lib/util/package-rules.ts:197]: store `sourceUrl?.toLowerCase()` to a local const, so we don't need to do the conversion on every loop

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/util/package-rules.ts (hunk 1 above)
Comment (dario-alv) [reply lib/util/package-rules.ts:197]: Updating, just a sec.

Comment (viceice) [reply lib/util/package-rules.ts:197]: Please add a test too

Comment (HonkingGoose) [reply lib/util/package-rules.ts:197]: Also please make the following changes:

- Edit the PR title to something like `fix: matchSourceUrlPrefixes compares case insensitive`
- Put `Fixes #10962` as text in the top comment, by doing that, GitHub will close the mentioned issue automatically once your PR gets merged. 😉 Read the [GitHub docs, linking a pull request to an issue using a keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) to learn more.

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: lib/util/package-rules.ts [hunk 2]
```diff
@@ -193,8 +193,9 @@ function matchesRule(
     positiveMatch = true;
   }
   if (matchSourceUrlPrefixes.length) {
+    const lowerCaseSourceUrl = sourceUrl.toLowerCase();
     const isMatch = matchSourceUrlPrefixes.some((prefix) =>
-      sourceUrl?.startsWith(prefix)
+      lowerCaseSourceUrl?.startsWith(prefix.toLowerCase())
```
Comment (HonkingGoose) [inline lib/util/package-rules.ts:198]: I think @viceice still wanted to have a test???

The PR title and PR body are good now.

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: lib/util/package-rules.ts (hunk 2 above)
Comment (viceice) [reply lib/util/package-rules.ts:198]: Yes, please add a unit test so we don't silently loose this in future

**Comment by HonkingGoose** [comment]
@rarkins Do we no longer need this fix, or did you maybe press the wrong button?

**Pull request closed without merging**